src/FsCodec.NewtonsoftJson/BoxCodec.fs

```diff
@@ -0,0 +1,101 @@
+/// Fork of FsCodec.NewtonsoftJson.Codec intended to provide equivalent calls and functionality, without actually serializing/deserializing as JSON
+/// This is a useful facility for in-memory stores such as Equinox's MemoryStore as it enables you to
+/// - efficiently test behaviors from an event sourced decision processing perspective (e.g. with Property Based Tests)
+/// - without paying a serialization cost and/or having to deal with sanitization of generated data in order to make it roundtrippable through same
+namespace FsCodec.Box
+
+open System
+open System.Runtime.InteropServices
+
+/// Provides Codecs that extract the Event bodies from a Union, using the conventions implied by using <c>TypeShape.UnionContract.UnionContractEncoder</c>
+/// If you need full control and/or have have your own codecs, see <c>FsCodec.Codec.Create</c> instead
+/// See <a href=""https://github.com/eiriktsarpalis/TypeShape/blob/master/tests/TypeShape.Tests/UnionContractTests.fs"></a> for example usage.
+type Codec private () =
+
+    /// Generate a <code>IUnionEncoder</code> Codec that roundtrips events by holding the boxed form of the Event body.
+    /// Uses <c>up</c> and <c>down</c> functions to facilitate upconversion/downconversion
+    ///   and/or surfacing metadata to the Programming Model by including it in the emitted <c>'Union</c>
+    /// The Event Type Names are inferred based on either explicit <c>DataMember(Name=</c> Attributes, or (if unspecified) the Discriminated Union Case Name
+    /// <c>Contract</c> must be tagged with </c>interface TypeShape.UnionContract.IUnionContract</c> to signify this scheme applies.
+    static member Create<'Union,'Contract,'Meta,'Context when 'Contract :> TypeShape.UnionContract.IUnionContract>
+        (   /// Maps from the TypeShape <c>UnionConverter</c> <c>'Contract</c> case the Event has been mapped to (with the raw event data as context)
+            /// to the <c>'Union</c> representation (typically a Discriminated Union) that is to be presented to the programming model.
+            up : FsCodec.ITimelineEvent<obj> * 'Contract -> 'Union,
+            /// Maps a fresh Event resulting from a Decision in the Domain representation type down to the TypeShape <c>UnionConverter</c> <c>'Contract</c>
+            /// The function is also expected to derive a <c>meta</c> object that will be held alongside the data (if it's not <c>None</c>)
+            ///   and an Event Creation <c>timestamp</c>.
+            down : 'Context option * 'Union -> 'Contract * 'Meta option * string * string * DateTimeOffset option,
```

Comment:
what are the 2 strings that are also part of this tuple? Would representing this in a type make it a clearer?

Reply:
this is the lowest level version available; the other calls delegate to this one - the two strings are the `correlationId` and `causationId` - in most instances where this is a relevant overload, you'd likely implement the whole interface directly.
(in `FsCodec.Codec.Create` I have this overload `private`; the only reason I havent done that here is for symmetry with `FsCodec.NewtonsoftJson.Codec.Create`, which exposes it [and we're trying to provide a drop-in replacement])

Reply:
updated comment to make the purpose clearer (I missed it in the first edit - it was never intended to be 'obvious' !)